File: PHP/Memory.php
<?php

    class Memory extends Hardware
    {
        private $memorySize = 0;
        private $frequency = 0;
        private $supportCUDA = "";

        function __construct(){
        }

        function setMemory($memorySize)
        {
            $this->memorySize = $memorySize;
        }

        function getMemory()
        {
            return $this->memorySize;
        }

        function setFrequency($frequency)
        {
            $this->frequency = $frequency;
        }

        function getFrequency()
        {
            return $this->frequency;
        }

        function setCUDA($supportCUDA)
        {
            $this->supportCUDA = $supportCUDA;
        }

        function getCUDA()
        {
            return $this->supportCUDA;
        }

        function __destruct(){
        }
    }

?>

File: CPP/Memory.cpp
#include <bits/stdc++.h>
#include "Hardware.cpp"
using namespace std;

class Memory : public Hardware
{
    private:
        int memorySize;
        int frequency;
        string supportCuda;

    public:
        Memory(){}

        //GetSet untuk MemorySize
        void setMemory(int memorySize)
        {
            this->memorySize = memorySize;
        }

        int getMemory()
        {
            return this->memorySize;    
        }

        //Getset untuk Frequency
        void setFrequency(int frequency)
        {
            this->frequency = frequency;
        }

        int getFrequency()
        {
            return this->frequency;
        }

        //GetSet untuk SupportCUDA
        void setCUDA(string supportCuda)
        {
            this->supportCuda = supportCuda;
        }

        string getCUDA()
        {
            return this->supportCuda;
        }


        ~Memory(){}

};